TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/qual/QualHardware.java

```diff
@@ -1,32 +1,47 @@
 package org.firstinspires.ftc.teamcode.hardware.qual;
 
-import com.qualcomm.hardware.bosch.BNO055IMU;
 import com.qualcomm.hardware.bosch.JustLoggingAccelerationIntegrator;
+import com.qualcomm.hardware.rev.RevHubOrientationOnRobot;
 import com.qualcomm.robotcore.hardware.DcMotor;
 import com.qualcomm.robotcore.hardware.DcMotorEx;
-import com.qualcomm.robotcore.hardware.DcMotorSimple;
 import com.qualcomm.robotcore.hardware.HardwareMap;
+import com.qualcomm.robotcore.hardware.IMU;
 import com.qualcomm.robotcore.hardware.Servo;
+import android.util.Size;
 
 import org.firstinspires.ftc.robotcore.external.Telemetry;
 import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
 import org.firstinspires.ftc.robotcore.external.navigation.AngleUnit;
-import org.firstinspires.ftc.robotcore.external.navigation.AxesOrder;
-import org.firstinspires.ftc.robotcore.external.navigation.AxesReference;
-import org.firstinspires.ftc.robotcore.external.navigation.Orientation;
-//import org.firstinspires.ftc.teamcode.hardware.RI3W.vision.RI3WComputerVisionProcessor;
-//import org.firstinspires.ftc.vision.VisionPortal;
-import org.firstinspires.ftc.vision.VisionProcessor;
+
+import org.firstinspires.ftc.robotcore.external.navigation.DistanceUnit;
+import org.firstinspires.ftc.robotcore.external.navigation.YawPitchRollAngles;
+import org.firstinspires.ftc.teamcode.hardware.vision.ComputerVisionProcessor;
+import org.firstinspires.ftc.vision.VisionPortal;
+import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor;
+import org.openftc.easyopencv.OpenCvCamera;
+import org.openftc.easyopencv.OpenCvCameraRotation;
 //import org.openftc.easyopencv.OpenCvCamera;
 //import org.openftc.easyopencv.OpenCvCameraRotation;
 
-public class QualHardware {
+public class QualHardware implements OpenCvCamera.AsyncCameraOpenListener {
     //Math for drive motor encoders
     public static final double TICKS_PER_MOTOR_REV = 537.7;
-    public static final double DRIVE_GEAR_RATIO = 1.0/1.0;
+    public static final double DRIVE_GEAR_RATIO = 1.0;
     public static final double TICKS_PER_DRIVE_REV = TICKS_PER_MOTOR_REV * DRIVE_GEAR_RATIO;
     public static final double WHEEL_CIRCUMFERENCE = Math.PI * 3.78;
     public static final double TICKS_PER_INCH = TICKS_PER_DRIVE_REV / WHEEL_CIRCUMFERENCE;
+    public static final double DRAGGER_DOWN_POSITION = .45;
+    public static final double DRAGGER_UP_POSITION = .7;
+    public enum DraggerPosition {UP, DOWN}
+    public DraggerPosition draggerPosition = DraggerPosition.UP;
+    static final double FX = 1442.66;
+    public static final double FY = 1442.66;
+    public static final double CX = 777.52;
+    public static final double CY = 162.257;
+    public OpenCvCamera camera;
+    public VisionPortal visionPortal;
+    public ComputerVisionProcessor propDetectionProcessor;
```

Comment:
When should the camera be turned off? One way to avoid the crash is to stop streaming after some criteria (such as, made a decision on the prop position).